### PR TITLE
add resource requests/limits per OSD-2109

### PR DESCRIPTION
**Pull request by bmeng**
current resource usage:
gcp-project-operator-756d6f99dc-l9wmx   4m           71Mi

### CODE REVIEW

File: deploy/operator.yaml [hunk 1]
```diff
@@ -20,6 +20,13 @@ spec:
           command:
           - gcp-project-operator
           imagePullPolicy: Always
+          resources:
+            requests:
+              cpu: "10m"
+              memory: "100Mi"
+            limits:
+              cpu: "50m"
+              memroy: "200Mi"
```
Comment (drpaneas) [inline deploy/operator.yaml:29]:
```suggestion
              memory: "200Mi"
```

**Comment by bmeng** [comment]
updated, thanks!

**Comment by drewandersonnz** [comment]
@bmeng can you explain why you've chosen request/limit ratio of 5x for CPU and 2x for RAM? 

This is something our customer have been doing and it has caused immense pain for them. 

I think we should specify 1.0x ratio (requests and limits match), unless there is a very good reason to do this differently.

**Comment by bmeng** [comment]
> @bmeng can you explain why you've chosen request/limit ratio of 5x for CPU and 2x for RAM?
> 
> This is something our customers have been doing and it has caused immense pain for them.
> 
> I think we should specify 1.0x ratio (requests and limits match), unless there is a very good reason to do this differently.

I do not have a good reason for this, basically I'd like to keep both cpu and memory have the same ratio 2x to make sure the service will not be in problem in peak.
But as the current usage of the cpu is too low, I am afraid that the 2x for limit of request will not be safe.

@drewandersonnz What is your suggestion of setting both the requests and limits here?

File: deploy/operator.yaml [hunk 2]
```diff
@@ -20,6 +20,13 @@ spec:
           command:
           - gcp-project-operator
           imagePullPolicy: Always
+          resources:
+            requests:
+              cpu: "10m"
```
Comment (NautiluX) [inline deploy/operator.yaml:25]:
```suggestion
              cpu: "500m"
```

File: deploy/operator.yaml [hunk 3]
```diff
@@ -20,6 +20,13 @@ spec:
...
+              memory: "100Mi"
```
Comment (NautiluX) [inline deploy/operator.yaml:26]:
```suggestion
              memory: "500Mi"
```

File: deploy/operator.yaml [hunk 4]
```diff
@@ -20,6 +20,13 @@ spec:
...
+            limits:
+              cpu: "50m"
```
Comment (NautiluX) [inline deploy/operator.yaml:28]:
```suggestion
              cpu: "500m"
```

File: deploy/operator.yaml [hunk 5]
```diff
@@ -20,6 +20,13 @@ spec:
...
+              memory: "200Mi"
```
Comment (NautiluX) [inline deploy/operator.yaml:29]:
```suggestion
              memory: "500Mi"
```

**Pull request merged**